### PR TITLE
Node 16 (cherry-pick) Upgrade and maxBuffer Fix

### PR DESCRIPTION
**Pull request by ozanovus**
**Task name**: AzureCLIV1 and AzureCLIV2

**Description**: "azure-pipelines-task-lib" library with "^4.0.0-preview" version has a problem with maxBuffer setting for external process call and this was causing problems for long outputs (such a large subscription list while login etc.)

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [X] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [X] Checked that applied changes work as expected

AzureCLI v1:

> Before upgrade (reproduce with "external" call):
> image
> 
> After upgrade:
> `"azure-pipelines-task-lib": "^4.0.1-preview"`
> image

AzureCLI v2:
> Before upgrade (reproduce with "external" call):
image
> After upgrade:
> `"azure-pipelines-task-lib": "^4.0.1-preview"`
image

Tested again with `"azure-pipelines-task-lib": "^4.1.0"`

image

### CODE REVIEW

**Review by max-zaytsev — APPROVED** [review]

File: Tasks/AzureCLIV1/package.json [hunk 1]
```diff
@@ -17,8 +17,8 @@
   },
   "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
   "dependencies": {
-    "azure-pipelines-task-lib": "^3.0.6-preview.0",
-    "@types/node": "^10.17.0",
+    "azure-pipelines-task-lib": "^4.0.1-preview",
```
Comment (DmitriiBobreshev) [inline Tasks/AzureCLIV1/package.json:20]: please update azure-pipeline-task-lib to latest version(4.1.0) according to [guide](https://github.com/microsoft/azure-pipelines-tasks/blob/master/docs/migrateNode16.md#upgrading-tasks-to-node-16)

File: Tasks/AzureCLIV2/package.json [hunk 2]
```diff
@@ -17,9 +17,9 @@
   },
   "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
   "dependencies": {
-    "@types/node": "^10.17.0",
+    "@types/node": "^16.11.39",
     "@types/q": "1.0.7",
-    "azure-pipelines-task-lib": "^3.1.2",
+    "azure-pipelines-task-lib": "^4.0.1-preview",
```
Comment (DmitriiBobreshev) [inline Tasks/AzureCLIV2/package.json:22]: same here

File: Tasks/AzureCLIV1/package.json (hunk 1 above)
Comment (ozanovus) [reply Tasks/AzureCLIV1/package.json:20]: @max-zaytsev , are we safe with version 4.1.0?

Comment (DmitriiBobreshev) [reply Tasks/AzureCLIV1/package.json]: @ozanovus latest version of task-lib has [the same maxBuffer](https://github.com/microsoft/azure-pipelines-task-lib/blob/master/node/toolrunner.ts#L597) value as 4.0.1-preview

Comment (ozanovus) [reply Tasks/AzureCLIV1/package.json]: thanks, updated.

File: Tasks/AzureCLIV2/package.json (hunk 2 above)
Comment (ozanovus) [reply Tasks/AzureCLIV2/package.json]: updated.

**Pull request merged**